app/src/main/java/http_server/ResponseBuilder.java

```diff
@@ -0,0 +1,58 @@
+package http_server;
+
+public class ResponseBuilder {
+
+  private final ResponseHandler handler;
+
+  public ResponseBuilder(ResponseHandler requestHandler) {
+    this.handler = requestHandler;
+  }
+
+  public Response buildResponse(RequestReader request) {
+    String path = request.getPath();
+    String method = request.getMethod();
+
+    if (path != null && method != null) {
+      switch (path) {
+        case "/simple_get_with_body":
+          return handler.buildSimpleGetWithBodyResponse();
+
+        case "/simple_get":
+          if (method.equals("HEAD")) {
+            return handler.buildHeadResponse();
+          }
+          return handler.buildSimpleGetResponse();
+
+        case "/echo_body":
+          if ("POST".equals(method)) {
+            String body = request.getBody();
+            if (body != null) {
```

Comment:
something to think about for a potential refactor if there's time in the future: I'm wondering if there's a way we can refactor this `buildResponse` method so there aren't so many nested conditional statements.  At this level for this particular line, we have 4 nested conditions.  Again, something to think about for the future.

Reply:
I suspect this refactor will occur in the ticket I am working on right now (Error Handling ticket) since the nested conditionals are mostly due to a lack of router functionality.